src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/Pkcs9ExtensionRequest.cs

```diff
@@ -19,34 +22,12 @@ private static byte[] EncodeAttribute(IEnumerable<X509Extension> extensions)
             if (extensions == null)
                 throw new ArgumentNullException(nameof(extensions));
 
-            // extensionRequest ATTRIBUTE ::= {
-            //  WITH SYNTAX ExtensionRequest
-            //  SINGLE VALUE TRUE
-            //  ID pkcs-9-at-extensionRequest
-            // }
-            //
-            // ExtensionRequest ::= Extensions
-            //
-            // Extensions  ::=  SEQUENCE SIZE (1..MAX) OF Extension
-            //
-            // Extension  ::=  SEQUENCE  {
-            //   extnID      OBJECT IDENTIFIER,
-            //   critical    BOOLEAN DEFAULT FALSE,
-            //   extnValue   OCTET STRING  }
+            X509ExtensionAsn[] extensionsAsn = extensions.Where(e => e != null).Select(e => new X509ExtensionAsn(e)).ToArray();
 
-            List<byte[][]> encodedExtensions = new List<byte[][]>();
-
-            foreach (X509Extension extension in extensions)
+            using (AsnWriter writer = AsnSerializer.Serialize(extensionsAsn, AsnEncodingRules.DER))
             {
-                if (extension == null)
-                    continue;
-
-                encodedExtensions.Add(extension.SegmentedEncodedX509Extension());
+                return writer.Encode();
```

Comment:
I'd write this as a foreach and avoid the double (or triple) copy in the LINQ statement.

```C#
using (AsnWriter writer = new AsnWriter(AsnEncodingRules.DER))
{
    writer.PushSequence();

    foreach (X509Extension e in extensions)
    {
        AsnSerializer.Serialize(new X509ExtensionAsn(e), writer);
    }

    writer.PopSequence();
    return writer.Encode();
}
```